string name
string type
byte[] data